zarr/core.py

```diff
@@ -1476,6 +1476,17 @@ def _set_basic_selection_zd(self, selection, value, fields=None):
         else:
             chunk[selection] = value
 
+        # clear chunk if it only contains the fill value
+        if np.all(np.equal(chunk, self._fill_value)):
```

Comment:
This seems like it could sometimes be an expensive operation. So it should be optional.

Reply:
How large are typical chunk sizes for you?

Reply:
regardless of chunk size, in many cases the user knows in advance whether this check will ever be useful. For dense imaging data, this conditional will almost never evaluate to true; for sparse data, this conditional will often evaluate to true. So the user should have some discretion over whether this chunk occurs.